src/files/XmlFile.spec.ts

```diff
@@ -524,24 +546,68 @@ describe('XmlFile', () => {
+            await testTranspile(trim`
                 <?xml version="1.0" encoding="utf-8" ?>
                 <component name="SimpleScene" extends="Scene">
                     <script type="text/brighterscript" uri="SimpleScene.bs"/>
                 </component>
-            `, `
+            `, trim`
                 <?xml version="1.0" encoding="utf-8" ?>
                 <component name="SimpleScene" extends="Scene">
-                    <script type="text/brightscript" uri="SimpleScene.brs"/>
+                    <script type="text/brightscript" uri="SimpleScene.brs" />
                     <script type="text/brightscript" uri="pkg:/source/lib.brs" />
                     <script type="text/brightscript" uri="pkg:/source/bslib.brs" />
                 </component>
             `, 'none', 'components/SimpleScene.xml');
         });
+
+        it('returns the XML unmodified if needsTranspiled is false', async () => {
+            let file = await program.addOrReplaceFile(
+                { src: s`${rootDir}/components/SimpleScene.xml`, dest: 'components/SimpleScene.xml' },
+                trim`
+                <?xml version="1.0" encoding="utf-8" ?>
+                <!-- should stay as-is -->
+                <component name="SimpleScene" extends="Scene" >
+                <script type="text/brightscript" uri="SimpleScene.brs"/>
+                </component>
+            `);
+            //prevent the default auto-imports to ensure no transpilation from AST
+            (file as any).getMissingImportsForTranspile = () => [];
+            expect(file.transpile().code).to.equal(trim`
+                <?xml version="1.0" encoding="utf-8" ?>
+                <!-- should stay as-is -->
+                <component name="SimpleScene" extends="Scene" >
+                <script type="text/brightscript" uri="SimpleScene.brs"/>
+                </component>
+            `);
+        });
+
+        it('needsTranspiled is false by default', async () => {
+            let file = await program.addOrReplaceFile(
+                { src: s`${rootDir}/components/SimpleScene.xml`, dest: 'components/SimpleScene.xml' },
+                trim`
+                <?xml version="1.0" encoding="utf-8" ?>
+                <component name="SimpleScene" extends="Scene" >
+                </component>
+            `);
+            expect(file.needsTranspiled).to.be.false;
+        });
+
+        it('needsTranspiled is true if an import is brighterscript', async () => {
+            let file = await program.addOrReplaceFile(
+                { src: s`${rootDir}/components/SimpleScene.xml`, dest: 'components/SimpleScene.xml' },
+                trim`
+                <?xml version="1.0" encoding="utf-8" ?>
+                <component name="SimpleScene" extends="Scene" >
+                    <script type="text/brightscript" uri="SimpleScene.bs"/>
+                </component>
+            `);
+            expect(file.needsTranspiled).to.be.true;
+        });
```

Comment:
There's a `bsconfig` flag called `autoImportComponentScript` which will automatically import a .brs or .bs script with the same name as the component, without needing to add a script tag. Can you add another test that verifies `needsTranspiled` is true when that flag is enabled, and `components/SimpleScene.bs` exists and is not directly referenced by a script tag?

Reply:
`needsTranspiled` isn't affected by the auto import - it's resolved for transpilation.

Reply:
the auto-import results in an additional `<script uri="path/to/codebehind/file.brs">` tag being injected into the XML at transpilie-time, which most definitely would require `needsTranspiled` to be true.

Reply:
Yes but this is resolved dynamically when transpiling as a result of calling `getMissingImportsForTranspile` - this forces transpilation.

Reply:
Oh. Ok, cool.